reactioncommerce/reaction
# feat: GraphQL address add mutation

Resolves #3919   
Impact: **minor**  
Type: **feature**

## Issue

Update the existing `addAccountAddressBookEntry` and fix any issues preventing it from working properly.

## Solution
Most of the implementation was already there. I made some updates to ensure it works as intended.

1. Updated `accounts/addressBookAdd` to return the added address
1. Allow accountId to be optional for `addAccountAddressBookEntry` mutation

## Breaking changes

Should be none; however,  the `accounts/addressBookAdd` was updated to return the added address. No methods depended on a specific return by that method, only that result was not undefined, which this satisfies.

## Testing

1. start the app and sign in
1. navigate to `localhost:3000/graphiql`
1. run a mutation, one example follows...

```
mutation {
    addAccountAddressBookEntry(input: {
    address: {
      fullName: "Mike M."
      country: "US"
      address1: "2110 min st"
      city: "Santa Monica"
      region: "CA"
      postal: "90405"
      isBillingDefault: true
      isShippingDefault: true
      isCommercial: true
      phone: "555-555-5555"
    }
  }) {
    clientMutationId,
    address {
      fullName
      address1
    }
}
}
```

---

1. With the app still running
1. Add an address from the profile screen to verify that it was added correctly
